zulipterminal/model.py

```diff
@@ -131,6 +131,26 @@ def __init__(self, controller: Any) -> None:
             for stream in self.stream_dict.values():
                 stream['date_created'] = None
 
+        # NOTE: The message_retention_days response has been added in
+        # server v3.0, feature level 17. For consistency we add this field
+        # on server iterations even before this with value of None.
+        def message_retention_days_response(days: int,
+                                            org_default: bool) -> str:
+            suffix = ' [Organization default]' if org_default else ''
+            return ('Indefinite' if days == -1 else str(days)) + suffix
+
+        for stream in self.stream_dict.values():
+            if feature_level is None or feature_level < 17:
+                stream['message_retention_days'] = None
+            else:
+                stream['message_retention_days'] = (
+                    message_retention_days_response(
+                        *(self.initial_data['realm_message_retention_days'],
+                          True) if stream['message_retention_days'] is None
+                        else (stream['message_retention_days'], False)
+                    )
+                )
```

Comment:
This is rather confusing to read. Do we need that nested function? The one-liner if/else is not easy to read.

Reply:
I'll fix this.
Just to clarify, do you want me to remove the whole function altogether, or just simplify the conditions for calling that function with those parameters?